tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts

```diff
@@ -123,10 +123,39 @@ export class CardFobControllerComponent {
     return this.axisDirection === AxisDirection.VERTICAL;
   }
 
+  private moveCurrentFob(
+    newStep: number,
+    timeSelection: TimeSelection
+  ): TimeSelection {
+    if (!this.timeSelection.end) {
+      timeSelection.start.step = newStep;
+      return timeSelection;
+    }
+
+    if (this.currentDraggingFob === Fob.END) {
+      if (newStep < this.timeSelection.start.step) {
+        timeSelection.end!.step = this.timeSelection.start.step;
+        this.currentDraggingFob = Fob.START;
+        return this.moveCurrentFob(newStep, timeSelection);
+      }
+      timeSelection.end!.step = newStep;
+
+      return timeSelection;
+    }
+
+    if (this.timeSelection.end && newStep > this.timeSelection.end.step) {
+      timeSelection.start.step = this.timeSelection.end.step;
+      this.currentDraggingFob = Fob.END;
+      return this.moveCurrentFob(newStep, timeSelection);
```

Comment:
Same here. I think it would be easy to just set timeSelection.end.step = newStep and return timeSelection.

Reply:
Recursion :skull: